sdk/src/connections/target/rgb_sensor.cpp

```diff
@@ -0,0 +1,663 @@
+        }
+    } else {
+        while (readBytes < size) {
+
+            if ((size - readBytes) >= CTRL_PACKET_SIZE) {
+                extCtrl.size = 2048 * sizeof(unsigned short);
+                extCtrl.p_u16 =
+                    (unsigned short *)((char *)firmware + readBytes);
+                extCtrl.id = V4L2_CID_AD_DEV_SET_CHIP_CONFIG;
+                memset(&extCtrls, 0, sizeof(struct v4l2_ext_controls));
+                extCtrls.controls = &extCtrl;
+                extCtrls.count = 1;
+
+                if (xioctl(dev->sfd, VIDIOC_S_EXT_CTRLS, &extCtrls) == -1) {
+                    LOG(WARNING)
+                        << "Programming AFE error "
+                        << "errno: " << errno << " error: " << strerror(errno);
+                    return Status::GENERIC_ERROR;
+                }
+                readBytes += CTRL_PACKET_SIZE;
+                usleep(100);
+            } else {
+                memset(buf, 0, CTRL_PACKET_SIZE);
+                memcpy(buf, ((const char *)firmware + readBytes),
+                       size - readBytes);
+                extCtrl.size = 2048 * sizeof(unsigned short);
+                extCtrl.p_u16 = (unsigned short *)buf;
+                extCtrl.id = V4L2_CID_AD_DEV_SET_CHIP_CONFIG;
+                memset(&extCtrls, 0, sizeof(struct v4l2_ext_controls));
+                extCtrls.controls = &extCtrl;
+                extCtrls.count = 1;
+
+                if (xioctl(dev->sfd, VIDIOC_S_EXT_CTRLS, &extCtrls) == -1) {
+                    LOG(WARNING)
+                        << "Programming AFE error "
+                        << "errno: " << errno << " error: " << strerror(errno);
+                    return Status::GENERIC_ERROR;
+                }
+                readBytes += CTRL_PACKET_SIZE;
+                usleep(100);
+            }
+        }
+    }
+
+    return status;
+}
+
+aditof::Status RgbSensor::getFrame(uint16_t *buffer,
+                                   aditof::BufferInfo *bufferInfo) {
+    using namespace aditof;
+    struct v4l2_buffer buf[m_implData->numVideoDevs];
+    struct VideoDev *dev;
+    Status status;
+
+    if (buffer == nullptr) {
+        LOG(ERROR) << "Received buffer null pointer";
+        return Status::INVALID_ARGUMENT;
+    }
+
+    for (unsigned int i = 0; i < m_implData->numVideoDevs; i++) {
+        dev = &m_implData->videoDevs[i];
+        status = waitForBufferPrivate(dev);
+        if (status != Status::OK) {
+            return status;
+        }
+
+        status = dequeueInternalBufferPrivate(buf[i], dev);
+        if (status != Status::OK) {
+            return status;
+        }
+        unsigned int buf_data_len;
+        uint8_t *pdata[m_implData->numVideoDevs];
+        status = getInternalBufferPrivate(&pdata[i], buf_data_len, buf[i], dev);
+        if (status != Status::OK) {
+            return status;
+        }
+
+        int k = 0;
+        for (unsigned int j = 0; j < (buf_data_len); j = j + 2) {
```

Comment:
no need for round brackets around buf_data_len

Reply:
done